carbon-design-system/carbon-addons-iot-react
# Card title resize

Closes #3266

**Summary**

This PR fixes a bug in the logic for `useHasTextOverflow` hook and updates the card to handle the changes from this fix.

**Change List (commits, features, bugs, etc)**

- Updated the `useHasTextOverflow.jsx` to make more dynamic
- Updated `Card.jsx` to handle changes need for text overlfow
- Wrote some jest unit test to test if it would change via prop update
- Wrote cypress test since our jest test manually update the values for `clientWidth` and `scrollWidth` to ensure it behaves well in actual browser. 

**Acceptance Test (how to verify the PR)**

- Go to 
[this story](https://deploy-preview-1824--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-card--with-ellipsed-title-tooltip-external-tooltip)
- click on subtitle and see that there is a tooltip
- click on isExpanded knob
- Check that tooltip is off for subtitle
then:
- Click on title and see tooltip
- Update the title in knobs to something short
- click on title and see tooltip has gone away


**Things to look for during review**
